### ISSUE TITLE
Unified project system

### ISSUE THREAD

**Issue by mkArtakMSFT** [dotnet/aspnetcore#46398]
To be filled in later

**Comment by javiercn** [comment]
https://github.com/dotnet/aspnetcore/issues/47885
https://github.com/dotnet/aspnetcore/issues/47884
https://github.com/dotnet/aspnetcore/issues/47883

**Comment by SteveSandersonMS** [comment]
Also copying a comment from https://github.com/dotnet/aspnetcore/issues/48758:

We need to update the new project template and SDK rules so that:

 * There are good conventions for including/excluding things in the Server and WebAssembly builds based on:
   * Directory
   * Filename
   * `#if`
 * We help developers not to include code or other assets like config in the WebAssembly compilation by mistake
   * For example, we might *only* include `.cs` files from a `WebAssembly` or `Shared` directory
   * For project/package references, we might have special metadata for "server", "webassembly", and "shared". Then if you add any project/package reference and it doesn't have this metadata (which is the case by default) then:
     * During development this logs a warning
     * During publish it's an error
     * ... so that developers aren't prevented from following existing tutorials (especially important for learners) but they do get a reasonable warning that they have to take some action before publishing to express a clear intent

**Comment by guimabdo** [comment]
Hi! I started testing and loving the new Blazor since RC1.

I noticed this unified experience is not in .net8 milestone anymore.

Will we be able to create a single project with Server and Client elements, at least in a preliminary way with .NET8?
- maybe with some experimental tool?
- Or only in .NET9?
- Or has this not been decided yet?

Thanks!

---
FYI, I'd like to structure something like this:

- My.Project.csproj
  - Server/ (server only)
    - Components/
      - HomePage.razor
  - WebAssembly/
    -  Components/
       - PeoplePage.razor (Auto)
  - Application/
    - People/
      - ListPeopleHandler.server.cs (server only)
      - ListPeopleHandler.client.cs (client only)
      - IListPeopleHandler.cs (both)
      - PersonDto.cs (both)

**Comment by SteveSandersonMS** [comment]
@guimabdo No, because of clear community feedback in https://github.com/dotnet/aspnetcore/issues/49079, we decided to focus on the multiproject solution for .NET 8 (that is, when you need both Server and WebAssembly build outputs). We are still open to creating a single project option in .NET 9 or later, but there is no option for that in .NET 8.
